test/opt/test_def_use.cpp

```diff
@@ -1151,4 +1152,221 @@ TEST(DefUseTest, OpSwitch) {
   }
 }
 
+// Creates an |result_id| = OpTypeInt 32 1 instruction.
+ir::Instruction OpTypeIntInstruction(uint32_t result_id) {
+  return ir::Instruction(SpvOp::SpvOpTypeInt, 0, result_id,
+                         {ir::Operand(SPV_OPERAND_TYPE_LITERAL_INTEGER, {32}),
+                          ir::Operand(SPV_OPERAND_TYPE_LITERAL_INTEGER, {1})});
+}
+
+// Creates an |result_id| = OpConstantTrue/Flase |type_id| instruction.
+ir::Instruction ConstantBoolInstruction(bool value, uint32_t type_id,
+                                        uint32_t result_id) {
+  return ir::Instruction(
+      value ? SpvOp::SpvOpConstantTrue : SpvOp::SpvOpConstantFalse, type_id,
+      result_id, {});
+}
+
+// Creates an |result_id| = OpLabel instruction.
+ir::Instruction LabelInstruction(uint32_t result_id) {
+  return ir::Instruction(SpvOp::SpvOpLabel, 0, result_id, {});
+}
+
+// Creates an OpBranch |target_id| instruction.
+ir::Instruction BranchInstruction(uint32_t target_id) {
+  return ir::Instruction(SpvOp::SpvOpBranch, 0, 0,
+                         {
+                             ir::Operand(SPV_OPERAND_TYPE_ID, {target_id}),
+                         });
+}
+
+struct AnalyzeInstDefUseTestCase {
+  std::vector<ir::Instruction> insts;  // instrutions to be analyzed in order.
+  InstDefUse expected_define_use;      // expected analysis result.
+};
+
+using AnalyzeInstDefUseTest =
+    ::testing::TestWithParam<AnalyzeInstDefUseTestCase>;
+
+// Test the analyzing result for individual instructions.
+TEST_P(AnalyzeInstDefUseTest, Case) {
+  auto tc = GetParam();
+
+  // Analyze the instructions.
+  opt::analysis::DefUseManager manager(nullptr);
+  for (ir::Instruction& inst : tc.insts) {
+    manager.AnalyzeInstDefUse(&inst);
+  }
+
+  CheckDef(tc.expected_define_use, manager.id_to_defs());
+  CheckUse(tc.expected_define_use, manager.id_to_uses());
+}
+
+// clang-format off
+INSTANTIATE_TEST_CASE_P(
+    TestCase, AnalyzeInstDefUseTest,
```

Comment:
Since we use `AnalyzeInstDefUse` for `AnalyzeDefUse` internally and the later already gets fairly good coverage, I'm OK with this relatively small test suite for `AnalyzeInstDefUseTest`.

But I think we should add some cases for a `DefUseManager` initialized from a non-`nullptr` module with its `AnalyzeInstDefUse` subsequently?